src/SimulationMap.cu

```diff
@@ -107,6 +124,19 @@ __device__ SimulationMap::~SimulationMap()
 }
 
 
+__device__ int SimulationMap::find_face_index(Face *face) const
+{
+    for(int i = 0; i < polyhedron->n_of_faces; ++i)
+    {
+        if(face == &polyhedron->faces[i])
+        {
+            return i;
+        }
+    }
```

Comment:
In `Polyhedron` all the faces are stored as an array. I am not 100% sure, but I think it's possible to retrieve the index of a `Face` using pointer arithmetic, without a loop

Reply:
> this is not obvious and surely should be documented

Or rewritten if you don't do this in purpose